tests/app/views/test_frameworks.py

```diff
@@ -538,186 +539,211 @@ class TestGetFrameworkSuppliers(BaseApplicationTest):
+
+            # Supplier 4 and 9's agreements were put on hold (only 4 subsequently remained on hold)
+            for supplier_id in (4, 9,):
+                with freeze_time(datetime.datetime(2016, 11, 30, 12-(supplier_id//3))):
+                    response = self.client.post(
+                        '/agreements/{}/on-hold'.format(agreement_ids[supplier_id]),
+                        data=json.dumps({'updated_by': 'example'}),
+                        content_type='application/json'
+                    )
+                    assert response.status_code == 200, response.get_data(as_text=True)
+
+            # Suppliers 6-10 have been approved for countersignature
+            for supplier_id in range(6, 11):
+                with freeze_time(datetime.datetime(2016, 11, 30, 15-supplier_id)):
+                    response = self.client.post(
+                        '/agreements/{}/approve'.format(agreement_ids[supplier_id]),
+                        data=json.dumps({
+                            'updated_by': 'example',
+                            "agreement": {'userId': 123},
+                        }),
+                        content_type='application/json'
+                    )
+                    assert response.status_code == 200, response.get_data(as_text=True)
+
+            # Suppliers 7-10 have countersigned agreements
+            for supplier_id in range(7, 11):
+                with freeze_time(datetime.datetime(2016, 12, 25, 5+supplier_id)):
+                    response = self.client.post(
+                        '/agreements/{}'.format(agreement_ids[supplier_id]),
+                        data=json.dumps({
+                            'updated_by': 'example',
+                            'agreement': {
+                                'countersignedAgreementPath': 'path/to/countersigned{}.pdf'.format(supplier_id)
+                            }
+                        }),
+                        content_type='application/json'
+                    )
+                    assert response.status_code == 200, response.get_data(as_text=True)
 
     def test_list_suppliers_related_to_a_framework(self, live_g8_framework):
         with self.app.app_context():
             # One G7 supplier
             response = self.client.get('/frameworks/g-cloud-7/suppliers')
             assert response.status_code == 200
             data = json.loads(response.get_data())
-            assert len(data['supplierFrameworks']) == 1
+            assert tuple(sf["supplierId"] for sf in data["supplierFrameworks"]) == (0,)
             # Ten G8 suppliers
             response = self.client.get('/frameworks/g-cloud-8/suppliers')
             assert response.status_code == 200
             data = json.loads(response.get_data())
-            assert len(data['supplierFrameworks']) == 10
+            assert tuple(sf["supplierId"] for sf in data["supplierFrameworks"]) == (0, 1, 2, 10, 5, 9, 4, 8, 3, 7, 6)
```

Comment:
This feels like we could do with one comment somewhere here in the tests that describes what the expected order is, e.g. something like  "supplierFrameworks are returned in order of ID if they don't have a framework agreement returned, and from oldest to newest returned if they do".  Just to make it clear what the tests are checking.